packages/epo-widget-lib/src/widgets/SourceSelector/SourceSelector.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  FunctionComponent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useState,
+} from "react";
+import { Alert, Source } from "@/types/astro";
+import { Image } from "@rubin-epo/epo-react-lib/dist/types/image";
+import { useTranslation } from "react-i18next";
+import { IconComposer } from "@rubin-epo/epo-react-lib";
+import * as Styled from "./styles";
+import Points from "./Points";
+import Message from "./Message";
+import { getLinearScale } from "./utilities";
+
+interface BlinkConfig {
+  autoplay?: boolean;
+  loop?: boolean;
+  duration?: number;
+}
+
+interface Elapsed {
+  day: number;
+  hour: number;
+}
+
+interface SourceSelectorProps {
+  width?: number;
+  height?: number;
+  sources: Source[];
+  alerts?: Alert[];
+  selectedSource?: Source[];
+  images: Image[];
+  selectionCallback?: (data: Source[]) => void;
+  blinkConfig?: BlinkConfig;
+  color?: string;
+}
+
+const SourceSelector: FunctionComponent<SourceSelectorProps> = ({
+  width = 600,
+  height = 600,
+  selectedSource: preSelectedSource = [],
+  sources,
+  alerts,
+  images,
+  selectionCallback,
+  blinkConfig,
+  color,
+}) => {
+  const [selectedSource, setSelectedSource] =
+    useState<Source[]>(preSelectedSource);
+  const [isLoaded, setLoaded] = useState(false);
+  const [message, setMessage] = useState<ReactNode>();
+  const [isMessageVisible, setMessageVisible] = useState(false);
+  const [elapsed, setElapsed] = useState<Elapsed | null>(null);
+  const { t } = useTranslation();
+  const svgId = "sourceSelectorWidget";
+
+  useEffect(() => {
+    selectionCallback && selectionCallback(selectedSource);
+  }, [selectedSource]);
+
+  const findData = (data: Source[], id: string, type: string) =>
+    data.filter((d) => d.id === id && d.type === type);
+
+  const handleClick = (event: MouseEvent<SVGElement>) => {
+    const { target } = event;
+    const { id, type } = (target as SVGElement).dataset;
+
+    if (id && type) {
+      const isAlreadySelected = findData(selectedSource, id, type).length > 0;
+
+      if (!isAlreadySelected) {
+        const newSelect = findData(sources, id, type);
+        setSelectedSource((value) => value.concat(newSelect));
+        setMessage(
+          <>
+            <IconComposer icon="checkmark" />
+            {t("source_selector.messages.success")}
+          </>
+        );
+        setMessageVisible(true);
+      }
+    } else {
+      setMessage(t("source_selector.messages.failure"));
+      setMessageVisible(true);
+    }
+  };
+
+  const handleMessageChange = () => {
+    setMessageVisible(false);
+  };
+
+  const handleBlinkChange = (index: number) => {
```

Comment:
when the blinker changes, look to see if there are associated alerts and then calculate the elapsed time